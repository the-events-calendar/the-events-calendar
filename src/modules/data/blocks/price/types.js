/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_PRICE_COST = `${ PREFIX_EVENTS_STORE }/SET_PRICE_COST`;
export const SET_PRICE_SYMBOL = `${ PREFIX_EVENTS_STORE }/SET_PRICE_SYMBOL`;
export const SET_PRICE_CODE = `${ PREFIX_EVENTS_STORE }/SET_PRICE_CODE`;
export const SET_PRICE_POSITION = `${ PREFIX_EVENTS_STORE }/SET_PRICE_POSITION`;
