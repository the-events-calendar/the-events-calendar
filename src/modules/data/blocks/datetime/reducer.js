/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';
import moment from 'moment/moment';

/**
 * Internal dependencies
 */
import {
	globals,
	date,
	moment as momentUtil,
	time,
} from '@moderntribe/common/utils';
import * as types from './types';

const { parseFormats, toTime } = momentUtil;

export const defaultStateToMetaMap = {
	start: '_EventStartDate',
	end: '_EventEndDate',
	dateTimeSeparator: '_EventDateTimeSeparator',
	timeRangeSeparator: '_EventTimeRangeSeparator',
	allDay: '_EventAllDay',
	timeZone: '_EventTimezone',
};

export const setInitialState = ( data ) => {
	const { meta } = data;

	Object.keys( defaultStateToMetaMap ).forEach( ( key ) => {
		const metaKey = defaultStateToMetaMap[ key ];
		if ( meta.hasOwnProperty( metaKey ) ) {
			DEFAULT_STATE[ key ] = meta[ metaKey ];
		}
	} );

	const { start, end } = DEFAULT_STATE;

	DEFAULT_STATE.startTimeInput = toTime( parseFormats( start ) );
	DEFAULT_STATE.endTimeInput = toTime( parseFormats( end ) );
	DEFAULT_STATE.naturalLanguageLabel = date.rangeToNaturalLanguage( start, end );
	DEFAULT_STATE.multiDay = ! momentUtil.isSameDay( momentUtil.toMoment( start ), momentUtil.toMoment( end ) );
};

const defaultStartTime = globals.defaultTimes().start ? globals.defaultTimes().start : '08:00:00';
const defaultEndTime = globals.defaultTimes().end ? globals.defaultTimes().end : '17:00:00';
const defaultStartTimeSeconds = time.toSeconds( defaultStartTime, time.TIME_FORMAT_HH_MM_SS );
const defaultEndTimeSeconds = time.toSeconds( defaultEndTime, time.TIME_FORMAT_HH_MM_SS );

export const defaultStartMoment = moment().startOf( 'day' ).seconds( defaultStartTimeSeconds );
export const defaultEndMoment = moment().startOf( 'day' ).seconds( defaultEndTimeSeconds );

const defaultStartDateTime = momentUtil.toDateTime( defaultStartMoment );
const defaultEndDateTime = momentUtil.toDateTime( defaultEndMoment );

export const DEFAULT_STATE = {
	start: defaultStartDateTime,
	end: defaultEndDateTime,
	startTimeInput: momentUtil.toTime( defaultStartMoment ),
	endTimeInput: momentUtil.toTime( defaultEndMoment ),
	naturalLanguageLabel: date.rangeToNaturalLanguage( defaultStartDateTime, defaultEndDateTime ),
	dateTimeSeparator: globals.settings().dateTimeSeparator
		? globals.settings().dateTimeSeparator
		: __( '@', 'the-events-calendar' ),
	timeRangeSeparator: globals.settings().timeRangeSeparator
		? globals.settings().timeRangeSeparator
		: __( '-', 'the-events-calendar' ),
	allDay: false,
	multiDay: false,
	timeZone: globals.timezone().timeZone ? globals.timezone().timeZone : date.FORMATS.TIMEZONE.string,
	showTimeZone: false,
	isEditable: true,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_DATETIME_BLOCK_EDITABLE_STATE:
			return {
				...state,
				isEditable: action.payload.isEditable,
			};
		case types.SET_START_DATE_TIME:
			return {
				...state,
				start: action.payload.start,
			};
		case types.SET_END_DATE_TIME:
			return {
				...state,
				end: action.payload.end,
			};
		case types.SET_START_TIME_INPUT:
			return {
				...state,
				startTimeInput: action.payload.startTimeInput,
			};
		case types.SET_END_TIME_INPUT:
			return {
				...state,
				endTimeInput: action.payload.endTimeInput,
			};
		case types.SET_NATURAL_LANGUAGE_LABEL:
			return {
				...state,
				naturalLanguageLabel: action.payload.label,
			};
		case types.SET_ALL_DAY:
			return {
				...state,
				allDay: action.payload.allDay,
			};
		case types.SET_MULTI_DAY:
			return {
				...state,
				multiDay: action.payload.multiDay,
			};
		case types.SET_SEPARATOR_DATE:
			return {
				...state,
				dateTimeSeparator: action.payload.separator,
			};
		case types.SET_SEPARATOR_TIME:
			return {
				...state,
				timeRangeSeparator: action.payload.separator,
			};
		case types.SET_TIME_ZONE:
			return {
				...state,
				timeZone: action.payload.timeZone,
			};
		case types.SET_TIMEZONE_VISIBILITY:
			return {
				...state,
				showTimeZone: action.payload.show,
			};
		default:
			return state;
	}
};
