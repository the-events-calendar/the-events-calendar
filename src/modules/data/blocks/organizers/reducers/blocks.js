/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import * as types from './../types';

export const byId = ( state = {}, action ) => {
	switch ( action.type ) {
		case types.ADD_BLOCK_ORGANIZER:
			return {
				...state,
				[ action.payload.id ]: action.payload.organizer,
			};
		case types.REMOVE_BLOCK_ORGANIZER:
			return Object.keys( state ).reduce( ( newState, id ) => {
				if ( id === action.payload.id ) {
					return newState;
				}

				return {
					...newState,
					[ id ]: state[ id ],
				};
			}, {} );
		default:
			return state;
	}
};

export const allIds = ( state = [], action ) => {
	switch ( action.type ) {
		case types.ADD_BLOCK_ORGANIZER:
			return [ ...state, action.payload.organizer ];
		case types.REMOVE_BLOCK_ORGANIZER:
			return state.filter( ( organizer ) => organizer !== action.payload.organizer );
		default:
			return state;
	}
};

export default combineReducers( {
	byId,
	allIds,
} );
