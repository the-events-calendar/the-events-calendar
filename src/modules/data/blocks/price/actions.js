/**
 * Internal dependencies
 */
import * as types from './types';

export const setCost = ( cost ) => ( {
	type: types.SET_PRICE_COST,
	payload: {
		cost,
	},
} );

export const setPosition = ( position ) => ( {
	type: types.SET_PRICE_POSITION,
	payload: {
		position,
	},
} );

export const setSymbol = ( symbol ) => ( {
	type: types.SET_PRICE_SYMBOL,
	payload: {
		symbol,
	},
} );

export const setCode = ( code ) => ( {
	type: types.SET_PRICE_CODE,
	payload: {
		code,
	},
} );
