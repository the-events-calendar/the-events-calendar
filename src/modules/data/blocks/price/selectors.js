/**
 * External dependencies
 */
import { createSelector } from 'reselect';

export const getPriceBlock = ( state ) => state.events.blocks.price;

export const getPrice = createSelector(
	[ getPriceBlock ],
	( block ) => block.cost,
);

export const getSymbol = createSelector(
	[ getPriceBlock ],
	( block ) => block.symbol,
);

export const getPosition = createSelector(
	[ getPriceBlock ],
	( block ) => block.position,
);

export const getCode = createSelector(
	[ getPriceBlock ],
	( block ) => block.code,
);
