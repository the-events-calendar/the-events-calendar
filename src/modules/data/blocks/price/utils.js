export const getPosition = ( showBefore ) => ( showBefore ? 'prefix' : 'suffix' );
