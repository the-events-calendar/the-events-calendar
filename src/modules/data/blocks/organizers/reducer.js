/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import { classic, classicSetInitialState, blocks } from './reducers';

export const setInitialState = ( data ) => {
	classicSetInitialState( data );
};

export default combineReducers( {
	blocks,
	classic,
} );
