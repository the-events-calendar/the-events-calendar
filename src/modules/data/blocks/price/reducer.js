/**
 * Wordpress dependenciess
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { settings, priceSettings } from '@moderntribe/common/utils/globals';
import { string } from '@moderntribe/common/utils';
import * as types from './types';

const position = string.isTruthy( settings() && settings().reverseCurrencyPosition )
	? 'suffix'
	: 'prefix';

export const DEFAULT_STATE = {
	position: priceSettings() && priceSettings().defaultCurrencyPosition
		? priceSettings().defaultCurrencyPosition
		: position,
	symbol: priceSettings() && priceSettings().defaultCurrencySymbol
		? priceSettings().defaultCurrencySymbol
		: __( '$', 'the-events-calendar' ),
	code: priceSettings() && priceSettings().defaultCurrencyCode
		? priceSettings().defaultCurrencyCode
		: __( 'USD', 'the-events-calendar' ),
	cost: '',
};

export const defaultStateToMetaMap = {
	position: '_EventCurrencyPosition',
	symbol: '_EventCurrencySymbol',
	code: '_EventCurrencyCode',
	cost: '_EventCost',
};

export const setInitialState = ( data ) => {
	const { meta } = data;

	Object.keys( defaultStateToMetaMap ).forEach( ( key ) => {
		const metaKey = defaultStateToMetaMap[ key ];
		if ( meta.hasOwnProperty( metaKey ) ) {
			DEFAULT_STATE[ key ] = meta[ metaKey ];
		}
	} );
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_PRICE_COST:
			return {
				...state,
				cost: action.payload.cost,
			};
		case types.SET_PRICE_POSITION:
			return {
				...state,
				position: action.payload.position,
			};
		case types.SET_PRICE_SYMBOL:
			return {
				...state,
				symbol: action.payload.symbol,
			};
		case types.SET_PRICE_CODE:
			return {
				...state,
				code: action.payload.code,
			};
		default:
			return state;
	}
};
