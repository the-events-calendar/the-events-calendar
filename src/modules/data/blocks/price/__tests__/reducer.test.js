/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events/data/blocks/price';
import reducer, {
	DEFAULT_STATE,
	defaultStateToMetaMap,
	setInitialState,
} from '@moderntribe/events/data/blocks/price/reducer';

const data = {
	meta: {
		_EventCurrencyPosition: 'prefix',
		_EventCurrencySymbol: '€',
		_EventCurrencyCode: 'EUR',
		_EventCost: '15',
	},
};

describe( '[STORE] - Price reducer', () => {
	it( 'Should return the default state', () => {
		expect( reducer( undefined, {} ) ).toEqual( DEFAULT_STATE );
	} );

	it( 'Should set the cost', () => {
		expect( reducer( DEFAULT_STATE, actions.setCost( 10 ) ) ).toMatchSnapshot();
	} );

	it( 'Should set the symbol position', () => {
		expect( reducer( DEFAULT_STATE, actions.setPosition( 'prefix' ) ) ).toMatchSnapshot();
	} );

	it( 'Should set the cost symbol', () => {
		expect( reducer( DEFAULT_STATE, actions.setSymbol( '€' ) ) ).toMatchSnapshot();
	} );

	it( 'Should set the cost code', () => {
		expect( reducer( DEFAULT_STATE, actions.setCode( 'EUR' ) ) ).toMatchSnapshot();
	} );

	it( 'Should return the default state to meta map', () => {
		expect( defaultStateToMetaMap ).toMatchSnapshot();
	} );

	it( 'Should set the initial state', () => {
		setInitialState( data );
		expect( DEFAULT_STATE ).toMatchSnapshot();
	} );
} );
