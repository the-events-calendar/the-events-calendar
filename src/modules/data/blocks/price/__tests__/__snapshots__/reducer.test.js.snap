// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Price reducer Should return the default state to meta map 1`] = `
Object {
  "code": "_EventCurrencyCode",
  "cost": "_EventCost",
  "position": "_EventCurrencyPosition",
  "symbol": "_EventCurrencySymbol",
}
`;

exports[`[STORE] - Price reducer Should set the cost 1`] = `
Object {
  "code": "USD",
  "cost": 10,
  "position": "prefix",
  "symbol": "$",
}
`;

exports[`[STORE] - Price reducer Should set the cost code 1`] = `
Object {
  "code": "EUR",
  "cost": "",
  "position": "prefix",
  "symbol": "$",
}
`;

exports[`[STORE] - Price reducer Should set the cost symbol 1`] = `
Object {
  "code": "USD",
  "cost": "",
  "position": "prefix",
  "symbol": "€",
}
`;

exports[`[STORE] - Price reducer Should set the initial state 1`] = `
Object {
  "code": "EUR",
  "cost": "15",
  "position": "prefix",
  "symbol": "€",
}
`;

exports[`[STORE] - Price reducer Should set the symbol position 1`] = `
Object {
  "code": "USD",
  "cost": "",
  "position": "prefix",
  "symbol": "$",
}
`;
